rust/krb5/src/lib.rs

```diff
@@ -95,6 +113,21 @@ impl KrbContext {
             raw: principal,
         })
     }
+
+    /// Get the default realm configured for this context.
+    pub fn default_realm(&self) -> Result<DefaultRealm, Error> {
```

Comment:
WDYT of writing out the default kerberos realm into a dedicated file of the secret volume to make it easier for operators such as hdfs-op to pick it up?

Reply:
I'd rather not add redundancy there, but the current HDFS situation isn't great either...

Perhaps another solution would be to add a utility to sble-base or something that uses libkrb5 to read out this stuff? (Either just reading the default realm, or something slightly more correct like "canonicalize this Kerberos principal?")

Reply:
Yeah, sounds like a more complicated but also better option